mlflow/tracking/fluent.py

```diff
@@ -2216,10 +2216,13 @@ def _create_logged_model(
     """
     if source_run_id is None and (run := active_run()):
         source_run_id = run.info.run_id
+
     if experiment_id is None and (run := active_run()):
         experiment_id = run.info.experiment_id
     elif experiment_id is None:
-        experiment_id = _get_experiment_id()
+        experiment_id = _get_experiment_id() or (
+            get_run(source_run_id).info.experiment_id if source_run_id else None
+        )
```

Comment:
If `_get_experiment_id` is None, fallback to getting the experiment id from `source_run_id`. We normally don't reach the fallback.

Reply:
I'll add a test.